zivid_camera/include/zivid_camera.h

```diff
@@ -42,74 +43,59 @@ class ZividCamera
-  bool shouldPublishPoints() const;
+  bool shouldPublishPointsXYZ() const;
+  bool shouldPublishPointsXYZRGBA() const;
   bool shouldPublishColorImg() const;
   bool shouldPublishDepthImg() const;
+  bool shouldPublishSnrImg() const;
   std_msgs::Header makeHeader();
-  void publishPoints(const std_msgs::Header& header, const Zivid::PointCloud& point_cloud);
+  void publishPointCloudXYZ(const std_msgs::Header& header, const Zivid::PointCloud& point_cloud);
+  void publishPointCloudXYZRGBA(const std_msgs::Header& header, const Zivid::PointCloud& point_cloud);
   void publishColorImage(const std_msgs::Header& header, const sensor_msgs::CameraInfoConstPtr& camera_info,
                          const Zivid::PointCloud& point_cloud);
   void publishColorImage(const std_msgs::Header& header, const sensor_msgs::CameraInfoConstPtr& camera_info,
-                         const Zivid::Image<Zivid::RGBA8>& image);
+                         const Zivid::Image<Zivid::ColorRGBA>& image);
   void publishDepthImage(const std_msgs::Header& header, const sensor_msgs::CameraInfoConstPtr& camera_info,
                          const Zivid::PointCloud& point_cloud);
+  void publishSnrImage(const std_msgs::Header& header, const sensor_msgs::CameraInfoConstPtr& camera_info,
+                       const Zivid::PointCloud& point_cloud);
   sensor_msgs::CameraInfoConstPtr makeCameraInfo(const std_msgs::Header& header, std::size_t width, std::size_t height,
                                                  const Zivid::CameraIntrinsics& intrinsics);
 
-  template <typename ConfigType_>
-  class ConfigDRServer
-  {
-  public:
-    using ConfigType = ConfigType_;
-    template <typename ZividSettings>
-    ConfigDRServer(const std::string& name, ros::NodeHandle& nh, const ZividSettings& defaultSettings);
-    void setConfig(const ConfigType& cfg);
-    const ConfigType& config() const
-    {
-      return config_;
-    }
-    const std::string& name() const
-    {
-      return name_;
-    }
-
-  private:
-    std::string name_;
-    boost::recursive_mutex dr_server_mutex_;
-    dynamic_reconfigure::Server<ConfigType> dr_server_;
-    ConfigType config_;
-  };
-
-  using CaptureGeneralConfigDRServer = ConfigDRServer<CaptureGeneralConfig>;
-  using CaptureFrameConfigDRServer = ConfigDRServer<CaptureFrameConfig>;
-  using Capture2DFrameConfigDRServer = ConfigDRServer<Capture2DFrameConfig>;
+  using Capture3DSettingsController =
+      CaptureSettingsController<Zivid::Settings, SettingsConfig, SettingsAcquisitionConfig>;
+  using Capture2DSettingsController =
+      CaptureSettingsController<Zivid::Settings2D, Settings2DConfig, Settings2DAcquisitionConfig>;
 
   ros::NodeHandle nh_;
   ros::NodeHandle priv_;
   ros::Timer camera_connection_keepalive_timer_;
   CameraStatus camera_status_;
-  std::unique_ptr<CaptureGeneralConfigDRServer> capture_general_config_dr_server_;
-  bool use_latched_publisher_for_points_;
+  bool use_latched_publisher_for_points_xyz_;
```

Comment:
Is there a need to keep all these bools as member functions? I guess they're just needed in the constructor?

Reply:
They are used other places, in fex shouldPublishPointsXYZRGBA, when we check if we should create and publish the different topics.

Reply:
Yes, I guess it's necessary since we then should publish even without subscribers. 
An alternative would be to always publish, but I think it's nicer the way we have it now; avoiding to do the work in f.ex. `publishPointCloudXYZRGBA()` unless subscribed to or latched.

Reply:
Yeah it should be like it is now, since it is important for performance. We have a lot of different outputs (Z map, XYZ, XYZRGB etc) so the user can select what they want for their use case, without any performance hit (avoid copying unnecessary data and so on).